tools/proguard5.3.2/test/proguard/util/ArrayUtilTest.java

```diff
@@ -0,0 +1,614 @@
+package proguard.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ArrayUtilTest {
+
+	@Test
+	public void testEqualByte() {
+		Assert.assertFalse(ArrayUtil.equal(
+				new byte[]{3, 2, 1}, new byte[]{1, 2, 3}, 4));
+
+		Assert.assertTrue(ArrayUtil.equal(new byte[]{1, 2, 3},
+				new byte[]{1, 2, 3}, 3)
+		);
+		Assert.assertTrue(ArrayUtil.equal(new byte[]{1, 2, 3},
+				new byte[]{1, 2, 3, 4}, 3)
+		);
+	}
+
+	@Test
+	public void testEqualShort() {
+		Assert.assertFalse(ArrayUtil.equal(new short[]{3, 2, 1},
+				new short[]{1, 2, 3}, 4)
+		);
+
+		Assert.assertTrue(ArrayUtil.equal(new short[]{1, 2, 3},
+				new short[]{1, 2, 3}, 3)
+		);
+		Assert.assertTrue(ArrayUtil.equal(new short[]{1, 2, 3},
+				new short[]{1, 2, 3, 4}, 3)
+		);
```

Comment:
What happens if you you put the larger array first?
Also if the size is larger than expected? I mean does it crash or return true?

Reply:
yeah it just crashes